platform/frontend_connectors/router_v2.go

```diff
@@ -133,22 +133,31 @@ func ConfigureSearchRouterV2(cfg *config.QuesmaConfiguration, dependencies quesm
 		return HandleResolveIndex(ctx, req.Params["index"], sr, cfg.Elasticsearch)
 	})
 
+	router.Register(routes.IndexPatternPitPath, and(method("POST"), isNotTargetingInternalIndex(), matchedAgainstPattern(tableResolver)), func(ctx context.Context, req *quesma_api.Request, _ http.ResponseWriter) (*quesma_api.Result, error) {
```

Comment:
`isNotTargetingInternalIndex` and `matchedAgainstPattern` are really similar, but `isNotTargetingInternalIndex` doesn't use table resolver.

I think that you should make a variant of `matchedAgainstPattern` that extracts the index name from `req.Params["indexPattern"]` - that way the code will use table resolver, instead of a new codepath.

Reply:
Good point!. Would be good to use `matchedAgainstPattern` if that's possible

Reply:
yea that has been really tempting, though I really liked the verbosity in the router code and how explicit the matching rules look - so decided to keep this separate.

I don't have a strong preference either way - so knowing my stance feel free to decide :)